test/components/dragBoxLayerTests.ts

```diff
@@ -2,284 +2,272 @@
+        TestMethods.triggerFakeDragSequence(target, quarterPoint, halfPoint);
 
-      svg.remove();
-    });
+        assert.isTrue(callbackCalled, "the callback was called");
+        assert.deepEqual(receivedBounds.topLeft, quarterPoint, "top-left point was set correctly");
+        assert.deepEqual(receivedBounds.bottomRight, halfPoint, "bottom-right point was set correctly");
 
-    it("onDrag()", () => {
-      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-      let dbl = new Plottable.Components.DragBoxLayer();
-      dbl.renderTo(svg);
+        callbackCalled = false;
+        dbl.offDrag(callback);
 
-      let startPoint = {
-        x: SVG_WIDTH / 4,
-        y: SVG_HEIGHT / 4
-      };
-      let endPoint = {
-        x: SVG_WIDTH / 2,
-        y: SVG_HEIGHT / 2
-      };
+        TestMethods.triggerFakeDragSequence(target, quarterPoint, halfPoint);
+        assert.isFalse(callbackCalled, "the callback was detached from the dragoBoxLayer and not called");
+        svg.remove();
+      });
 
-      let receivedBounds: Plottable.Bounds;
-      let callbackCalled = false;
-      let callback = (b: Plottable.Bounds) => {
-        receivedBounds = b;
-        callbackCalled = true;
-      };
-      dbl.onDrag(callback);
+      it("calls the onDragEnd callback", () => {
+        dbl.renderTo(svg);
 
-      let target = dbl.background();
-      TestMethods.triggerFakeDragSequence(target, startPoint, endPoint);
+        let receivedBounds: Plottable.Bounds;
+        let callbackCalled = false;
+        let callback = (b: Plottable.Bounds) => {
+          receivedBounds = b;
+          callbackCalled = true;
+        };
+        dbl.onDragEnd(callback);
 
-      assert.isTrue(callbackCalled, "the callback was called");
-      assert.deepEqual(receivedBounds.topLeft, startPoint, "top-left point was set correctly");
-      assert.deepEqual(receivedBounds.bottomRight, endPoint, "bottom-right point was set correctly");
+        let target = dbl.background();
+        TestMethods.triggerFakeDragSequence(target, quarterPoint, halfPoint);
 
-      callbackCalled = false;
-      dbl.offDrag(callback);
+        assert.isTrue(callbackCalled, "the callback was called");
+        assert.deepEqual(receivedBounds.topLeft, quarterPoint, "top-left point was set correctly");
+        assert.deepEqual(receivedBounds.bottomRight, halfPoint, "bottom-right point was set correctly");
+        dbl.offDragEnd(callback);
+        callbackCalled = false;
 
-      TestMethods.triggerFakeDragSequence(target, startPoint, endPoint);
-      assert.isFalse(callbackCalled, "the callback was detached from the dragoBoxLayer and not called");
+        TestMethods.triggerFakeDragSequence(target, quarterPoint, halfPoint);
+        assert.isFalse(callbackCalled, "the callback was detached from the dragoBoxLayer and not called");
+        svg.remove();
+      });
 
-      svg.remove();
-    });
+      it("can register two callbacks for the same event", () => {
```

Comment:
in theory, each line of the code should be tested. (I don't think we have coverage test yet?)
right now each `onEvent` is an one-liner calling ` this._eventCallbacks.add(callback);`
what if someone makes a mistake by emptying `this._eventCallbacks` before adding for one of the events, it will not be caught if we only test multiple callback for one event